### PR TITLE
Feature/i2c slave

### PR DESCRIPTION
**Pull request by smeenka**
Implemented a blocking i2c_slave, and modified the i2c_master such that incorrect frame errors can be detected.
The master can cope with an ACK on the last received byte, but also with a NACK on the last received byte from the slave

Tests where the communication between master and slave is tested (good and bad cases) can be found here: https://github.com/smeenka/rust-examples

### CODE REVIEW

File: src/i2c.rs [hunk 1]
```diff
@@ -70,6 +100,25 @@ impl Config {
         };
         psc << 28 | scldel << 20 | sdadel << 16 | sclh << 8 | scll
     }
+    // slave address 1 as 7 bit address, in range 0 .. 127
+    pub fn slave_address(&mut self, own_address: u8) {
```
Comment (andresv) [inline src/i2c.rs:104]: those should be doc comments with `///`

**Comment by andresv** [comment]
Can you check that comments in traits and methods are actually doc comments with `///`, could you check also letter casings in there.

**Comment by dotcypress** [comment]
Looks great! 🦀

I will try to use this change on some of my i2c projects first, then I'm happy to merge this!

**Comment by smeenka** [comment]
This is a blocking i2c implementation so I used the Result value. For function slave_wait_addressed
the nb:Result would be more appropriate (because non-blocking). Please discuss if this is wanted, than I will make a next pull request with this change
Also really want to move i2c into a directory, and this implementation is subdir blocking. Then split in several files (master, slave, lib, config etc) like is done for serial. For now this is a too big change. Please discuss if this is wanted
As soon i2c is in a dir I can start creating a non-blocking variant, based on callbacks and/or with all nb:result variant.

**Comment by dotcypress** [comment]
@smeenka feel free to move i2c into a directory, this is great idea!
Also it would be cool to have both implementation: blocking and non-blocking.

I tried this branch on some of my projects - works fine. 🦀

**Comment by smeenka** [comment]
Note that the interfaces are found in i2c/mod.rs and the implementation in i2c/blocking.rs. In the application one has to use the i2c/blocking to include the implementation, See https://github.com/smeenka/rust-examples.

**Comment by dotcypress** [comment]
@smeenka lets fix code format issue and we good to merge this!

**Comment by smeenka** [comment]
Fixed the formatting issue

**Comment by dotcypress** [comment]
Thank you!

**Pull request merged**